### PR TITLE
Custom target filename

### PR DESCRIPTION
**Pull request by jupe**
## Status
**READY**

## Migrations
NO

## Description
introduce optional `--target_filename <filename>` -parameter

**NOTE:** implemented top of #145 

## Todos
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: mbed_flasher/main.py [hunk 1]
```diff
@@ -196,6 +196,9 @@ def argparser_setup(self, sysargs):
                                        'also multiple targets can be given. '
                                        'Short target_id matches boards by prefix',
                                   default=None, metavar='TARGET_ID', action='append')
+        parser_flash.add_argument('-f', '--target_filename',
```
Comment (juhhov) [inline mbed_flasher/main.py]: Maybe just --target_filename is enough?

Comment (jupe) [reply mbed_flasher/main.py]: true, 'll remove `-f` option

File: mbed_flasher/flashers/FlasherMbed.py [hunk 2]
```diff
@@ -85,7 +85,7 @@ def can_flash(target):
         return True
 
     # pylint: disable=too-many-return-statements, duplicate-except
-    def flash(self, source, target, method, no_reset):
+    def flash(self, source, target, method, no_reset, target_filename):
```
Comment (jupe) [inline mbed_flasher/flashers/FlasherMbed.py]: what do you think, is this "public API" ? I would say no... Note that this change is breaking change if we think about this is public API..

Comment (juhhov) [reply mbed_flasher/flashers/FlasherMbed.py]: It should not break if default value is given for the target_filename. When the value is default it should work as before this change.

Comment (juhhov) [reply mbed_flasher/flashers/FlasherMbed.py]: Anyway, this is not public API IMO either.

**Comment by jupe** [comment]
mysterious failure:
```
Couldn't parse '/builds/ws/her_PR-150-AIXXIRXBENT7E7B5KUWSC/mbed-flasher/sample.py' as Python source: 'invalid syntax' at line 5
```

PR does not touch that file...

**Comment by RomanSaveljev** [comment]
Why problem(s) does this PR solve?

**Comment by juhhov** [comment]
@jupe could this be merged?

File: mbed_flasher/main.py [hunk 3]
```diff
@@ -292,7 +295,11 @@ def subcmd_flash_handler(self, args):
             raise FlashError(message=msg,
                              return_code=EXIT_CODE_TARGET_ID_MISSING)
 
-        check_is_file_flashable(self.logger, args.input)
+        check_file(self.logger, args.target_filename or args.input)
+        check_file(self.logger, args.input)
+        check_file_exists(self.logger, args.input)
+        check_file_extension(self.logger, args.target_filename or args.input)
```
Comment (juhhov) [inline mbed_flasher/main.py]: Should both the input and target file name extensions be checked?

**Comment by juhhov** [comment]
Add test for cli usage as well. Missing also tests for invalid target filename values.

**Pull request merged**